lib/rubygems/defaults.rb

```diff
@@ -28,26 +28,26 @@ def self.default_spec_cache_dir
   # specified in the environment
 
   def self.default_dir
-    path = if defined? RUBY_FRAMEWORK_VERSION then
-             [
-               File.dirname(RbConfig::CONFIG['sitedir']),
-               'Gems',
-               RbConfig::CONFIG['ruby_version']
-             ]
-           elsif RbConfig::CONFIG['rubylibprefix'] then
-             [
-              RbConfig::CONFIG['rubylibprefix'],
-              'gems',
-              RbConfig::CONFIG['ruby_version']
-             ]
-           else
-             [
-               RbConfig::CONFIG['libdir'],
-               ruby_engine,
-               'gems',
-               RbConfig::CONFIG['ruby_version']
-             ]
-           end
+    path =  if defined? RUBY_FRAMEWORK_VERSION then
```

Comment:
This looks weird to me. Why the change? Is there a cop correcting this?

Reply:
I didn't do automated corrections. This was indented by an odd number of spaces originally, so both RuboCop and Vim were throwing a fit about it.

I personally prefer this style:

```
path =
    if ...
       ...
    end
```

But I for some reason didn't think to do that at the time.

Reply:
@deivid-rodriguez do you think I should switch to the approach of `path =` followed by the entire block indented two spaces? (See the above comment.) Alternatively, is there another way you'd format it?

Reply:
Hi @duckinator! I don't see the previous odd indentation that grants a reformat of the whol if-block. The only indentation problem there is the elseif block array, where the three elements of it are missing one space on their indentation.

Regarding the style you prefer, it's not my preferred one, but I don't mind if we adopted it. How common is it in the current code base? Is rubocop able to enforce it currently?

Reply:
@deivid-rodriguez the apparent missing spaces were because those array elements were indented an even multiple of two spaces, whereas the `if`/`elsif`/`end` statements (and the other array elements) were not.

e.g., here's that function [as it is on master](https://raw.githubusercontent.com/rubygems/rubygems/master/lib/rubygems/defaults.rb) with the lines annotated with how they're indented:

```
  def self.default_dir # 2 spaces
    path = if defined? RUBY_FRAMEWORK_VERSION then # 4 spaces
             [ # 13 spaces
               File.dirname(RbConfig::CONFIG['sitedir']), # 15 spaces
               'Gems', #15 spaces
               RbConfig::CONFIG['ruby_version'] # 15 spaces
             ] # 13 spaces
           elsif RbConfig::CONFIG['rubylibprefix'] then # 11 spaces
             [ # 13 spaces
              RbConfig::CONFIG['rubylibprefix'], # 14 spaces
              'gems', # 14 spaces
              RbConfig::CONFIG['ruby_version'] # 14 spaces
             ] # 13 spaces
           else # 11 spaces
             [ # 13 spaces
               RbConfig::CONFIG['libdir'], # 15 spaces
               ruby_engine, # 15 spaces
               'gems', # 15 spaces
               RbConfig::CONFIG['ruby_version'] # 15 spaces
             ] # 13 spaces
           end # 11 spaces

    @default_dir ||= File.join(*path) # 4 spaces
  end # 2 spaces
```

Note that `path = ` is 7 characters (including the space after the equal sign), which means aligning things relative to that will inherently leave indentation at an odd number of spaces.

Many editors, when presented with that and set up for 2-space indentation, will simply add a single space(!) when you hit tab.

So there's two approaches folks take here:

1. indent relative to the previous line.
2. indent to even numbers of spaces.

Both are completely valid, but I lean towards the latter because it avoids things like the unintended single-space-indentation in situations like we have here.

My preferred approach of adding a newline after the equal sign means that indentation is relative to the _start_ of the line, not the end, and thus sidesteps the entire problem.

Reply:
In this case the problematic lines are lines 10-12 that are indented only one space relative to the opening bracket of the array instead of two as they should.

I don't know about your editor configuration but it's the first time I hear of a coding style that enforces one or two spaces after the equal sign of an assignment depending on whether the resulting indentation is even or odd.

Regarding the other style (introduce a line break after the equal sign) and indent the if/case relative to the beginning of the previous line, that's something I've seen in some places but you're still going to need to fix the middle array indentation since the inconsistency will not be gone unless you fix it.

I would personally enable `Layout/IndentArray` _and_ `Layout/AlignArray` and  and auto-correct them via `rubocop --auto-correct`. I'm posting an [screenrecord](https://asciinema.org/a/isaFUwxjlupSDJWrykDHZtPua) for the motivation: enabling only one of them leaves the code in a weird style. I think this is a bug in rubocop which I'll report, but we can skip it by enabling both cops.

Reply:
Ok, so I checked the [documentation](https://github.com/rubocop-hq/rubocop/blob/master/manual/cops_layout.md#layoutindentarray), and it's super-clear about this. There's no bug in rubocop:

> This cop checks the indentation of the first element in an array literal where the opening bracket and the first element are on separate lines. The other elements' indentations are handled by the AlignArray cop.

Reply:
What I was trying to say is that with some editors, if e.g. you have tabs set to be 2 spaces and are indented 3 spaces, hitting tab will only take you to 4 spaces (not 5).

But if `rubocop --auto-correct` can resolve that then it's not much of a problem, I don't think.